ddtrace/ddtrace.go

```diff
@@ -171,6 +171,15 @@ type StartSpanConfig struct {
 	Context context.Context
 }
 
+func NewStartSpanConfig(opts ...StartSpanOption) StartSpanConfig {
```

Comment:
Let's add a doc string since this a public function. What it does is obvious but perhaps we should clarify why and how it can be useful compared to using start options.